src/collectors/windows.plugin/MonitorSQL.c

```diff
@@ -113,16 +113,19 @@ static ULONGLONG netdata_MSSQL_fill_long_value(SQLHSTMT *stmt, const char *mask,
 
 void dict_mssql_fill_instance_transactions(struct mssql_db_instance *mdi)
 {
+    if (unlikely(!mdi || !mdi->parent || !mdi->parent->conn || !mdi->parent->conn->collect_buffer))
+        return;
+
     char object_name[NETDATA_MAX_INSTANCE_OBJECT + 1] = {};
     long value = 0;
     SQLLEN col_object_len = 0, col_value_len = 0;
 
-    SQLCHAR query[sizeof(NETDATA_QUERY_TRANSACTIONS_PER_INSTANCE_MASK) + NETDATA_MAX_INSTANCE_OBJECT + 1];
+    SQLCHAR query[sizeof(NETDATA_QUERY_BUFFER_STATS) + NETDATA_MAX_INSTANCE_OBJECT + 1];
     snprintfz(
-        (char *)query,
-        sizeof(NETDATA_QUERY_TRANSACTIONS_PER_INSTANCE_MASK) + NETDATA_MAX_INSTANCE_OBJECT,
-        NETDATA_QUERY_TRANSACTIONS_PER_INSTANCE_MASK,
-        mdi->parent->instanceID);
+            (char *)query,
+            sizeof(NETDATA_QUERY_BUFFER_STATS) + NETDATA_MAX_INSTANCE_OBJECT,
+            NETDATA_QUERY_BUFFER_STATS,
+            mdi->parent->instanceID);
```

Comment:
For this query `NETDATA_QUERY_BUFFER_STATS` it does not seem to use the `instanceID`

Reply:
Last commit addressed this.